### DISCUSSION TITLE
How should I get environment variable in Kong Admin API (api.lua)

### DISCUSSION THREAD

**Discussion by mong0520** [Kong/kong#8502] [Help] [answered]
I am trying to develop a hello word level plugin with an admin api (listen on :8001) in `my-plugin/api.lua` and it works.
However, I can't read the value from environment variable in `api.lua` but works on plugin execution stage.

Please advise how should I read the env variable in `api.lua` ? I'm pretty sure I can read it from plugin execution level but not Admin API

code snippet of `api.lua` attached
```lua
...
["/test"] = {
    GET = function(self, db, helpers)
      return "hello, " .. (os.getenv("NAME")  or "nil")
    end
  },
...
``` 

<img width="604" alt="image" src="https://user-images.githubusercontent.com/10734784/156728658-be737997-468c-4330-a631-5bd96e2725dc.png">

**Comment by ADD-SP** [comment, marked as answer]
> By default, Nginx removes all environment variables inherited from its parent process except the TZ variable.
> https://nginx.org/en/docs/ngx_core_module.html#env

So if you want to get an environment variable, you can edit the `kong.conf`, write a line `nginx_main_env = HOME`, or set environment `KONG_NGINX_MAIN_ENV=HOME` before you start Kong. Now, you can get it in Lua `os.getenv("HOME")`.

**Reply by StarlightIbuki** [reply]
I am still confused that "I can read it from plugin execution level"
